bundles/pubsub/pubsub_admin_tcp/src/pubsub_tcp_handler.c

```diff
@@ -1266,9 +1269,12 @@ void pubsub_tcpHandler_handler(pubsub_tcpHandler_t *handle) {
         }
         pubsub_tcpHandler_close(handle, events[i].ident);
       } else if (events[i].flags & EV_ERROR) {
+        celixThreadRwlock_unlock(&handle->dbLock);
         L_ERROR("[TCP Socket]:EPOLLERR  ERROR read from socket %s\n", strerror(errno));
         pubsub_tcpHandler_close(handle, events[i].ident);
         continue;
+      } else {
+        celixThreadRwlock_unlock(&handle->dbLock);
```

Comment:
The unlock is always done, isn't it cleaner to move it before or after the if?

Reply:
The problem solved here is that pubsub_tcpHandler_close also locks a mutex and can cause potential deadlocks because of different ordering (I forgot where the other order happened)

So since there is no use in locking the dbLock while calling pubsub_tcpHandler_close, we unlock it before that happens. Hence the unlock is done in every branch. It's either that, or introduce a boolean variable to check if it has already been unlocked.

Reply:
But, since it is done in every branch, it can be done before the if?

Reply:
Hmm, I think the first unlock on line 1255 should be below what is now line 1257. I'll have to check again if that is possible, or if I ran into some issues.